arrays.dd

```diff
@@ -263,9 +266,12 @@ $(H2 $(LNAME2 array-setting, Array Setting))
 
 ---------
 int[3] s;
+int[] d = new int[3];
 int* p;
 
+d[] = 3;     // same as d[0] = 3, d[1] = 3, d[2] = 3
 s[] = 3;     // same as s[0] = 3, s[1] = 3, s[2] = 3
+s = 3;       // also works (because s is a static array)
```

Comment:
What I was trying to do here, is reduce the EXTREME confusion caused by the current documentation combined with the current semantics. I have noticed that the experienced members of this community don't generally see this problem anymore. I respectfully suggest that this problem - of the experienced members minimizing glaring inconsistencies in 'basic functionality' - is D's biggest threat. Here I'm focusing on the confusing differences in semantics between static and dynamic arrays.

The other areas I see this same threat are:
1) "dynamic arrays" which change their identity when you grow them.
2) the semantics of utf8 strings.